tools/chatgpt/classes/instance.php

```diff
@@ -72,9 +72,9 @@ protected function extend_store_formdata(stdClass $data): void {
         $this->set_endpoint($endpoint);
 
         $this->set_customfield2($enabled);
-        $this->set_customfield3($resourcename);
-        $this->set_customfield4($deploymentid);
-        $this->set_customfield5($apiversion);
+        $this->set_customfield3(trim($resourcename));
```

Comment:
I don't think this is the right place to do this.

This should be done in https://github.com/mebis-lp/moodle-local_ai_manager/blob/00c7d2c0bbe6e66fecfc809e245c7f6d3d493063/classes/local/aitool_option_azure.php#L87

Same goes for the aitool_option_temperature class: https://github.com/mebis-lp/moodle-local_ai_manager/blob/00c7d2c0bbe6e66fecfc809e245c7f6d3d493063/classes/local/aitool_option_temperature.php#L103

Also we should properly trim all the inputs in https://github.com/mebis-lp/moodle-local_ai_manager/blob/00c7d2c0bbe6e66fecfc809e245c7f6d3d493063/classes/base_instance.php#L518 as well as all the inputs in the different `extend_store_formdata` methods in the tool instance classes.